### PR TITLE
[Docs] Add initial docs pass, adding lots of cross-reference links.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR does a generic pass over the docs:
- Adds cross-reference links where appropriate.
- Makes code blocks copyable.
- Minor grammar fixes/prose tweaks.

### Code Blocks with Copy-to-Clipboard Button

Screenshot 2023-06-06 at 4 25 29 PM

### Cross-reference Links

Screenshot 2023-06-06 at 4 25 57 PM

Screenshot 2023-06-06 at 4 26 34 PM

### CODE REVIEW

File: docs/source/10-min.ipynb [hunk 1]
```diff
@@ -92,15 +93,16 @@
    "source": [
     "You can also load DataFrames from other sources, such as:\n",
     "\n",
-    "1. CSV files: `daft.read_csv(\"s3://bucket/*.csv\")`\n",
-    "2. Parquet files: `daft.read_parquet(\"/path/*.parquet\")`\n",
-    "3. JSON line-delimited files: `daft.read_json(\"/path/*.parquet\")`\n",
-    "4. Files on disk: `daft.from_glob_path(\"/path/*.jpeg\")`\n",
+    "1. CSV files: {func}`daft.read_csv(\"s3://bucket/*.csv\") <daft.read_csv>`\n",
```
Comment (samster25) [inline docs/source/10-min.ipynb:96]: What does this look like in the notebook?

File: docs/source/learn/user_guides/read-write.rst [hunk 2]
```diff
@@ -49,7 +49,7 @@ However, if instead you are reading a set of files that are not container file f
     #   ...
 
 
-This is especially useful for reading things such as a folder of images or documents into Daft. A common pattern is to then download data from these files into your DataFrame as bytes, using the ``Expression.url.download()`` method.
+This is especially useful for reading things such as a folder of images or documents into Daft. A common pattern is to then download data from these files into your DataFrame as bytes, using the :meth:`.url.download() <daft.expressions.expressions.ExpressionUrlNamespace.download>` method.
```
Comment (samster25) [inline docs/source/learn/user_guides/read-write.rst:52]: Haha that's a long path!

**Review by samster25 — APPROVED** [review]
LGTM, thanks for doing this 😃

File: docs/source/10-min.ipynb (hunk 1 above)
Comment (clarkzinzow) [reply docs/source/10-min.ipynb:96]: A bit less readable in-notebook, but we ultimately decided that the cross-referencing was worth it when writing up a bunch of examples for Ray's ML ecosystem. Ideally there would be a Jupyter plugin that would render the markdown properly in the notebook itself, but we never figured that out.

Screenshot 2023-06-06 at 6 53 54 PM

Comment (samster25) [reply docs/source/10-min.ipynb:96]: I'll let @jaychia plug in on this since he wrote it!

Comment (clarkzinzow) [reply docs/source/10-min.ipynb:96]: @jaychia Any opposition to this? I'd like to merge this soon if possible!

Comment (clarkzinzow) [reply docs/source/10-min.ipynb:96]: Talked to @jaychia offline and he says that he thinks this is a net-positive change! Merging now.

Comment (jaychia) [reply docs/source/10-min.ipynb:96]: Ship!! 🚢

**Pull request merged**